packages/ui/src/validators/providers/utils.ts

```diff
@@ -123,14 +124,16 @@ export const getValidatorInfo = (
 
           const commission = validator.commission
           const averageReward = rewards.reduce((sum, reward) => sum.add(reward.eraReward), BN_ZERO).divn(rewards.length)
-          const apr =
-            Number(
-              averageReward
-                .muln(ERAS_PER_YEAR)
-                .muln(100 - commission)
-                .muln(100)
-                .div(staking.total)
-            ) / 100
+          const yearlyReward = averageReward.muln(ERAS_PER_YEAR)
+          const afterCommission = yearlyReward.muln(100 - commission).divn(100)
+          const aprValue = afterCommission.muln(100).div(staking.total)
+          const maxSafeValue = new BN(1000000)
+
+          if (aprValue.gt(maxSafeValue)) {
+            return { APR: 1000000 }
+          }
+
+          const apr = aprValue.toNumber()
```

Comment:
The result in pioneer LGTM!
For code simplicity: 
- Are there cases where apr exceeds 1M?
- Is 1M the highest safe value?
- Isn't `aprValue` always a number?